src/kiara_plugin/topic_modelling/modules/corpus_preparation.py

```diff
@@ -30,83 +25,221 @@ def create_inputs_schema(self):
             },
             "file_name_col": {
                 "type": "string",
-                "doc": "The column containing file names with metadata. In order to work, file names need to comply with LCCN pattern '/sn86069873/1900-01-05/' containing publication reference and date."
+                "doc": "The column containing file names with metadata. In order to work, file names need to comply with LCCN pattern '/sn86069873/1900-01-05/' containing publication reference and date.",
             },
             "map": {
                 "type": "list",
                 "doc": "List of lists of unique publications references and publication names in the collection provided in the same order.",
                 "optional": True,
-            }
+            },
         }
 
     def create_outputs_schema(self):
         return {
             "corpus_table": {
                 "type": "table",
-                "doc": "Augmented table containing extracted metadata."
+                "doc": "Augmented table containing extracted metadata.",
             }
         }
 
     def process(self, inputs, outputs) -> None:
-        
+        import re
+        import polars as pl
+        import pyarrow as pa
+
         table_obj = inputs.get_value_obj("corpus_table")
         column_name = inputs.get_value_obj("file_name_col").data
-        pub_refs, pub_names = None, None
 
-        try:
-            pub_refs = inputs.get_value_obj("map").data[0]
-            pub_names = inputs.get_value_obj("map").data[1]
+        if not column_name:
```

Comment:
This test probably makes sense, because the input string could be empty. So leave it in, but it makes me wonder when an empty string will ever be useful input compared to just no input...

Reply:
I reproduced the process from here (e.g. lines 222/223) since this module seems to have been reviewed by you https://github.com/DHARPA-Project/kiara_plugin.tropy/blob/develop/src/kiara_plugin/tropy/modules/create.py

Write
Preview
Cancel
Update comment